### ISSUE TITLE
[BUG] empty_like can generate a column with an invalid col_name pointer

### ISSUE THREAD

**Issue by jlowe** [rapidsai/cudf#1904]
**Describe the bug**
`cudf::empty_like` creates an uninitialized `gdf_column` and initializes it by calling `gdf_column_view_augmented`, but this leaves the `col_name` field uninitialized.  If the uninitialized data happens to not be zero then code can crash trying to access a bad column name pointer or corrupt the heap if the seemingly valid pointer is freed.

**Expected behavior**
`cudf::empty_like` should generate a column with a null `col_name`.

**Comment by ayushdg** [comment]
A part of the changes in #1542 should handle this. Specifically this bit: https://github.com/rapidsai/cudf/pull/1542/files#diff-61b7dd2a4e4fe523938574222b14a385R213

**Comment by harrism** [comment]
Setting the pointer to `nullptr` in `gdf_column_view`/`gdf_column_view_augmented` is a trivial fix, anyone can feel free to submit a PR.

**Comment by jlowe** [comment]
> Setting the pointer to nullptr in gdf_column_view/gdf_column_view_augmented is a trivial fix

Note that if we go this route then `gdf_column_wrapper` in io/utilities/wrapper_utils.hpp will break unless they are updated.  They fill in the col_name field _before_ calling `gdf_column_view_augmented`:
```c++
  gdf_column_wrapper(gdf_size_type size, gdf_dtype dtype,
                     gdf_dtype_extra_info dtype_info, const std::string name) {
    col = (gdf_column *)malloc(gdf_column_sizeof());
    col->col_name = (char *)malloc(name.length() + 1);
    std::strcpy(col->col_name, name.c_str());
    gdf_column_view_augmented(col, nullptr, nullptr, size, dtype, 0, dtype_info);
  }
```

**Comment by jlowe** [comment]
Looks like #1914 will fix this by zero-initializing the gdf_column when it's created.

**Comment by jlowe** [comment]
Fixed by #1914.
